10_temperature_object/temperature.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+class Temperature
+  def initialize(hash)
+    @ftemp = hash[:f]
+    @ctemp = hash[:c]
+    @ftemp = (@ctemp.to_f * 9 / 5) + 32 unless @ctemp.nil?
+    @ctemp = (@ftemp - 32) * 5 / 9 unless @ftemp.nil?
+  end
+
+  def in_fahrenheit
+    @ftemp
+  end
+
+  def in_celsius
+    @ctemp
+  end
+
+  def self.from_celsius(temp)
+    self.new(:c => temp)
+  end
+
+  def self.from_fahrenheit(temp)
+    self.new(:f => temp)
+  end
+end
+
+class Celsius < Temperature
+  def initialize(temp)
+    @ctemp = temp
+    @ftemp = (@ctemp.to_f * 9 / 5) + 32 unless @ctemp.nil?
+  end
+end
+
+class Fahrenheit < Temperature
+  def initialize(temp)
+    @ftemp = temp
+    @ctemp = (@ftemp - 32) * 5 / 9 unless @ftemp.nil?
```

Comment:
Do you think duplicating logic is a good solution?

Reply:
Nope :(

Reply:
Didn't found a better solution for duplicated code used in subclasses :/

Reply:
Temperature.from_celsius needs to return Celsius instance.
Temperature.from_fahrenheit needs to return Fahrenheit instance.